File: ARC/ARC029/arc029a.py
n = int(input())
li = [int(input()) for _ in range(n)]
res = 99999
for i in range(n ** 2):
    t = i
    plateA = 0
    plateB = 0
    for j in range(n):
        if t & 1 == 1:
            plateA += li[j]
        else:
            plateB += li[j]
        t >>= 1
    res = min(res, max(plateA, plateB))
print(int(res))

File: ARC/ARC029/arc029a.rb
n = gets.to_i
li = []
n.times { li.push gets.to_i }
maxnum = 999999
1.upto(2 ** n) { |i|
	t = i
	plateA = 0
	plateB = 0
	0.upto(n - 1) { |j|
		if (t & 1) == 1
			plateA += li[j]
		else
			plateB += li[j]
		end
		t >>= 1
	}
	maxnum = [maxnum, [plateA, plateB].max].min
}
puts maxnum